servlet/src/main/java/io/undertow/servlet/core/ErrorPages.java

```diff
@@ -61,10 +61,13 @@ public String getErrorLocation(final Throwable exception) {
         if (location == null && exception instanceof ServletException) {
             Throwable rootCause = ((ServletException) exception).getRootCause();
             //Iterate through any nested JasperException in case it is in JSP development mode
-            while (rootCause != null && rootCause instanceof ServletException) {
+            while (rootCause != null && rootCause instanceof ServletException && location == null) {
+                for (Class c = rootCause.getClass(); c != null && location == null; c = c.getSuperclass()) {
+                    location = exceptionMappings.get(c);
+                }
                 rootCause = ((ServletException) rootCause).getRootCause();
             }
-            if (rootCause != null) {
+            if (rootCause != null && location == null) {
                 for (Class c = rootCause.getClass(); c != null && location == null; c = c.getSuperclass()) {
```

Comment:
we don't need to check for location equals null any longer because of the change in the previous line

Reply:
@aogburn  I'll take the freedom to update the PR myself because I'm working on releasing and tagging Undertow with this fix. Thanks for your PR!

Reply:
@aogburn never mind, my bad, this code is actually correct and does not need to be updated